azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java

```diff
@@ -105,10 +105,10 @@
       final String location = template.getLocation().getId();
       final int[] inboundPorts = template.getOptions().getInboundPorts();
 
-      final String storageAccountName = templateOptions.getStorageAccountName().get();
-      final String virtualNetworkName = templateOptions.getVirtualNetworkName().get();
-      final String reservedIPAddress = templateOptions.getReservedIPName().orNull();
-      final String subnetName = templateOptions.getSubnetName().get();
+      final String storageAccountName = templateOptions.getStorageAccountName();
+      final String virtualNetworkName = templateOptions.getNetworks().isEmpty() ? null : Iterables.getOnlyElement(templateOptions.getNetworks());
```

Comment:
It does not seem OK to fail if there is more than one network provided. If the current implementation only allows to attach the VM to one network, then fail throwing an appropriate message so users know what's going on.
In any case, if we support the `networks` option we should allow to attach the VM to all provided networks.